File: dat/tmp.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <iterator>
#include <sstream>
#include <math.h>
#include <stdlib.h>
using namespace std;

const double H_PI = 2 * M_PI;

double fRand(double fMin, double fMax)
{
  double f = (double)rand() / RAND_MAX;
  return fMin + f * (fMax - fMin);
}

template <size_t N>
void splitString(string (&arr)[N], string str)
{
  int n = 0;
  istringstream iss(str);
  for (auto it = istream_iterator<string>(iss); it != istream_iterator<string>() && n < N; ++it, ++n)
    arr[n] = *it;
}
int main()
{
  ofstream fout;
  ifstream fin;
  string line;
  string arr[5];
  string src = "instance100.txt";
  string dest = "instance100.js";
  fout.open(dest);
  fin.open(src);
  fout << "const dat = [\n";
  string data;
  getline(fin, data);
  for (int i = 0; i < 1500; i++)
  {
    getline(fin, data);
    splitString(arr, data);
    fout << "{\npos:{\nx:" << (arr[0]) << ",\ny:" << arr[1] << "\n},\nbeta:" << arr[3] << ",\n},\n";
  }
  fout << "]";
  fin.close();
  fout.close();
  return 0;
}

File: dat/tmp.js
import * as fs from 'fs'
var src = 'instance100.txt'
var dest = 'instance100.js'
var dat = []
fs.readFile(src,'utf-8',function(err,data){
  if(err) console.log(err)
  dat = data.split('\r\n')
})
var writeln = 'const dat = [\n'
for(let i=1;i<1501;i++){
  var nmb = dat[i].split(' ')
  console.log(nmb);
  writeln += '{\npos:{\nx:' + nmb[0] + ',\ny:' + nmb[1] + '\n},\nbeta:' + nmb[3]+',\n},\n'
}
writeln += ']'
fs.writeFile(dest,writeln,(err)=>{
  if(err) console.log(err)
  else console.log('done')
})